crates/debug/src/transform/expression.rs

```diff
@@ -376,9 +421,29 @@ pub fn compile_expression<R>(
 where
     R: Reader,
 {
+    let mut jump_arcs: HashMap<usize, usize> = HashMap::new();
     let mut pc = expr.0.clone();
+
     let buf = expr.0.to_slice()?;
     let mut parts = Vec::new();
+    macro_rules! push {
+        ($unread:expr, $part:expr) => {{
+            let part = $part;
+            if let (CompiledExpressionPart::Code(cc2), Some(CompiledExpressionPart::Code(cc1))) =
+                (&part, parts.last())
+            {
+                let mut combined = cc1.clone();
+                parts.pop();
```

Comment:
is `combined` == to result of `parts.pop()`? so you don't have to clone

Reply:
What I'd really like to do is:
``` rust
            if let (CompiledExpressionPart::Code(cc2), Some(CompiledExpressionPart::Code(cc1))) =
                (&part, parts.last())
            {
		cc1.extend_from_slice(cc2);
            } else {
                parts.push(CompiledExpressionPart::LandingPad {
                    original_pos: (expr.0.len().into_u64() - $unread) as usize,
                });
                parts.push(part)
            }
```
But for that `Code` needs to carry a mutable vector.
I tried various other ways of getting rid of the `.clone()` but I guess my Rust-fu is not strong enough for that. I'd love to see your idea written out though. Indeed the `.pop()` removes the `Code` which has `cc1` in it, and then it gets appended by `cc2` and. pushed back as a fresh `Code`.

Reply:
You are not wrong above, just use `parts.last_mut()` instead of `.last()`.

Reply:
Success!